llmfoundry/models/inference_api_wrapper/openai_causal_lm.py

```diff
@@ -228,9 +228,15 @@ def process_result(self, completion: Optional['ChatCompletion']):
             tensors = []
             for t in self.tokenizer(
                     completion.choices[0].message.content)['input_ids']:
-                tensors.append(
-                    self.tokenizer.construct_logit_tensor(
-                        {self.tokenizer.decode([t]): 0.0}))
+
+                # Construct tensor of shape (vocab_size,) with logprobs for each token
+                tokenizer_logprobs = {self.tokenizer.decode([t]): 0.0}
+                tensor = torch.tensor([min(tokenizer_logprobs.values()) - 1] *
```

Comment:
I know all this is taken from tiktoken.py but this logic doesn't make sense to me.
L233: set each log prob to 0 in a dict
L234/235: take minimum of all tokenizer_logprobs (we set them all to 0) and subtract 1 to get -1, duplicate that value (always -1?) to make a tensor of size len(tokenizer). 
L236-238: at tensor location for each token in tokenizer_logprobs, set probability to 0.0 instead of -1.

So we're not returning tru logprobs at all if I'm not mistaken?

Reply:
@bmosaicml

Reply:
[Silliest refactor ever](https://github.com/mosaicml/llm-foundry/pull/914/commits/861a2bd66ee2cc1c79d1e38c11b32d3de7b5714b), but this should make this significantly clearly

Reply:
OpenAI only returns top 5 logprobs, but there's still `len(tokenizer) - 5 logprobs` that need to be mocked in order for the metric's math to be computed properly. n.b. recall that the metrics expect to be able to index into a tensor of size len(tokenizer) and inspect the logprob there